006-strimzi-ui.md

```diff
@@ -0,0 +1,178 @@
+
+Additional Kafka capabilities which could be added later include (but would not be limited to):
+
+- Create, update and delete topics on the topics list page (recontextualised as a Topics page - as per mock up)
+- View Consumer group status for a given topic, or the whole cluster
+- View the Brokers in a given cluster, their configuration, and where appropriate, allow modification of broker configuration
+- Provide details of Kafka listener/bootstrap addresses, along with sample configuration, to allow streamline client connectivity
+
+If desired, additional Strimzi provided capabilities, such as Kafka Users, could be leveraged/added/managed through this UI also (for example, allowing access to the UI and its capabilities, general CRUD updates to them etc). 
+
+Finally, one capability which may be of interest (which may have bearing on how this is surfaced in Strimzi itself) is having a single UI, which can manage multiple Kafka clusters deployed via the Strimzi operator.
+
+In all of these cases, capabilities can be added in a prioritised order, and should be added in a phased manner themselves (for example, add a view of the brokers and their configuration, then the ability to modify select configuration).
+
+#### Proposed deployment
+
+This UI could be deployed as a part of Strimzi as follows:
+
+![Suggested deployment](./images/006-deployment.png)
+
+Where:
+
+- The UI is deployed standalone, akin to how the Kafka bridge, Kafka Connect are currently for example (so a new CRD would be required, with configuration needed to reference which cluster(s) to interact with).
+- The UI's spec would contain 0 to N Kafka clusters the UI will operate against, which will: 
+  - Contain 1 or more 'backend' entries. These represent backend sources of data which can be surfaced in the UI.
+    - One of these entries will be the `strimzi-api` (ie each Kafka kind will contain its own api server).
+  - Each of these entries will define items such as the address to use to connect to it, any auth credentials required, Transport security required, etc.
+- The UI's spec will contain general configuration for the UI - such as certificates to expose to clients etc.
+
+This could look as follows in a CR configured by a user (note that common fields such as image, or readiness and liveness probes have been omitted here for conciseness, but would be present in a full CR). Comments are included inline for clarification of intent/type:
+
+```
+...
+spec:
+  clusters: # Array/list
+  - name: 'dev-cluster' # string - metadata, but could be used in the UI (eg for cluster selection if more than one cluster) in the clusters array (order of which respected by the UI)
+    clientCert: <mounted TLS cert to serve to clients> # optional - if omitted, UI server will serve via http rather than https
+    uiConfig: <stringified JSON, containing general UI config - eg features to disable/enable, logging> # string - optional - if provided will override default configuration values
```

Comment:
what about a reference to a ConfigMap (containing the JSON) instead of having a potential huge JSON here?

Reply:
Perhaps we allow for an inline JSON string, or a ConfigMap + Key reference.

Reply:
I think a config map makes sense - what I would like to avoid is filling the CRD/CR with UI config options, which normally a user would not need to change. WIll update to reflect this